ra2ce/analyses/indirect/traffic_analysis/traffic_analysis_base.py

```diff
@@ -0,0 +1,159 @@
+"""
+                    GNU GENERAL PUBLIC LICENSE
+                      Version 3, 29 June 2007
+
+    Risk Assessment and Adaptation for Critical Infrastructure (RA2CE).
+    Copyright (C) 2023 Stichting Deltares
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from abc import abstractmethod
+import abc
+import itertools
+import operator
+from typing import Any
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import ast
+from ra2ce.analyses.indirect.traffic_analysis.accumulated_traffic_dataclass import (
+    AccumulatedTraffic,
+)
+
+
+class TrafficAnalysisBase(abc.ABC):
+    gdf: gpd.GeoDataFrame
+    od_table: gpd.GeoDataFrame
+    destinations_names: str
+
+    def optimal_route_od_link(
+        self,
+    ) -> pd.DataFrame:
+        """
+        Gets the optimal routes based on regular, egalitarian and prioritarian traffic.
+
+        Returns:
+            pd.DataFrame: Datafarme with the traffic indices for each of analysis.
+        """
+        origin_nodes = np.unique(self.gdf["origin"])
+        destination_nodes = np.unique(self.gdf["destination"])
+
+        unique_destination_nodes = np.unique(list(self.od_table["d_id"].fillna("0")))
+        count_destination_nodes = len([x for x in unique_destination_nodes if x != "0"])
+
+        _traffic: dict[str, AccumulatedTraffic] = {}
+        for o_node in origin_nodes:
+            for d_node in destination_nodes:
+                opt_path = self._get_opt_path_values(o_node, d_node)
+                for u_node, v_node in itertools.pairwise(opt_path):
+                    _nodes_key_name = self._get_node_key(u_node, v_node)
+                    _calculated_traffic = self._calculate_origin_node_traffic(
+                        o_node, count_destination_nodes
+                    )
+                    if "," in d_node:
+                        _calculated_traffic *= len(d_node.split(","))
+
+                    _accumulated_traffic = _traffic.get(
```

Comment:
How calling this method jumps into the __add__() function of the AccumulatedTraffic?